projects/packages/publicize/src/jetpack-social-settings/class-settings.php

```diff
@@ -16,55 +16,37 @@
  * This class is used to get and update Jetpack_Social_Settings.
  * Currently supported features:
  *      - Social Image Generator
- *      - Auto Conversion
  */
 class Settings {
 	/**
 	 * Name of the database option.
 	 *
 	 * @var string
 	 */
-	const OPTION_PREFIX = 'jetpack_social_';
-	// cSpell:ignore AUTOCONVERT
-	const AUTOCONVERT_IMAGES       = 'autoconvert_images';
+	const OPTION_PREFIX            = 'jetpack_social_';
 	const IMAGE_GENERATOR_SETTINGS = 'image_generator_settings';
 
 	const DEFAULT_IMAGE_GENERATOR_SETTINGS = array(
 		'enabled'  => false,
 		'template' => Templates::DEFAULT_TEMPLATE,
 	);
 
-	const DEFAULT_AUTOCONVERT_IMAGES_SETTINGS = array(
-		'enabled' => true,
-	);
-
 	/**
 	 * Migrate old options to the new settings. Previously SIG settings were stored in the
-	 * jetpack_social_image_generator_settings option. Now they are stored in the jetpack_social_settings
-	 * together with the auto conversion settings.
+	 * jetpack_social_image_generator_settings option. Now they are stored in the jetpack_social_settings.
 	 *
 	 * TODO: Work out if this is possible on plugin upgrade
 	 *
 	 * @return void
 	 */
 	private function migrate_old_option() {
-		// Migrating from the old option.
-		$old_auto_conversion_settings = get_option( 'jetpack_social_settings' );
-		if ( ! empty( $old_auto_conversion_settings ) ) {
-			update_option( self::OPTION_PREFIX . self::AUTOCONVERT_IMAGES, array( 'enabled' => ! empty( $old_auto_conversion_settings['image'] ) ) );
-			delete_option( 'jetpack_social_settings' );
-		}
-		// Checking if the new option is valid.
-		$auto_conversion_settings = get_option( self::OPTION_PREFIX . self::AUTOCONVERT_IMAGES );
-		// If the option is not set, we don't need to delete it.
-		// If it is set, but it is not an array or it does not have the enabled key, we delete it.
-		if ( false !== $auto_conversion_settings && ( ! is_array( $auto_conversion_settings ) || ! isset( $auto_conversion_settings['enabled'] ) ) ) {
-			delete_option( self::OPTION_PREFIX . self::AUTOCONVERT_IMAGES );
-		}
+		// Delete the old options.
+		delete_option( 'jetpack_social_settings' );
+		delete_option( 'jetpack_social_autoconvert_images' );
```

Comment:
This will cause problems because `delete_option` always runs an uncached get before deleting, and `migrate_old_options` runs on every `get_settings` call. We have had issues because of this before. We have to make sure we only run `delete_option` if necessary, in this case, if the option is not empty as it was before.

Reply:
Oh! Thank you for catching that.

Reply:
Fixed. Thanks